src/eth_provider/provider.rs

```diff
@@ -450,7 +471,66 @@ where
         ))
     }
 
-    async fn call(&self, request: TransactionRequest, block_id: Option<BlockId>) -> EthProviderResult<Bytes> {
+    async fn call(
+        &self,
+        request: TransactionRequest,
+        block_id: Option<BlockId>,
+        state_overrides: Option<StateOverride>,
+        block_overrides: Option<Box<BlockOverrides>>,
+    ) -> EthProviderResult<Bytes> {
+        // Create the EVM overrides from the state and block overrides.
+        let evm_overrides = EvmOverrides::new(state_overrides, block_overrides);
+
+        // Check if either state_overrides or block_overrides is present.
+        if evm_overrides.has_state() || evm_overrides.has_block() {
+            // Create the configuration environment with the chain ID.
+            let cfg_env = CfgEnv::default().with_chain_id(self.chain_id().await?.unwrap_or_default().to());
+
+            // Retrieve the block header details.
+            let Header { number, timestamp, miner, base_fee_per_gas, difficulty, .. } =
+                self.header(&block_id.unwrap_or_default()).await?.unwrap_or_default();
+
+            // Create the block environment with the retrieved header details and transaction request.
+            let block_env = BlockEnv {
+                number: U256::from(number.unwrap_or_default()),
+                timestamp: U256::from(timestamp),
+                gas_limit: U256::from(request.gas.unwrap_or_default()),
+                coinbase: miner,
+                basefee: U256::from(base_fee_per_gas.unwrap_or_default()),
+                prevrandao: Some(B256::from_slice(&difficulty.to_be_bytes::<32>()[..])),
+                ..Default::default()
+            };
+
+            // Combine the configuration environment with the handler configuration.
+            let cfg_env_with_handler_cfg =
+                CfgEnvWithHandlerCfg { cfg_env, handler_cfg: HandlerCfg::new(SpecId::CANCUN) };
+
+            // Create a snapshot of the Ethereum database using the block ID.
+            let mut db = EthCacheDatabase(CacheDB::new(EthDatabase::new(self, block_id.unwrap_or_default())));
+
+            // Prepare the call environment with the transaction request, gas limit, and overrides.
+            let env = prepare_call_env(
+                cfg_env_with_handler_cfg,
+                block_env,
+                request.clone(),
+                request.gas.unwrap_or_default().try_into().expect("Gas limit is too large"),
+                &mut db.0,
+                evm_overrides,
+            )?;
+
+            // Execute the transaction using the configured EVM.
+            let res = tokio::task::block_in_place(|| {
```

Comment:
why block in place here? is this needed?

Reply:
just fixed you are right